moveit_ros/planning/planning_pipeline/include/moveit/planning_pipeline/planning_pipeline.h

```diff
@@ -57,6 +57,79 @@ namespace planning_pipeline
 class PlanningPipeline
 {
 public:
+  /**
+   * @brief      This class contains the rcl interfaces for easier testing
+   */
+  class MiddlewareHandle
+  {
+  public:
+    /**
+     * @brief      Destroys the object.
+     */
+    virtual ~MiddlewareHandle() = default;
+
+    virtual bool has_parameter(const std::string& name) const = 0;
+    virtual void get_parameter(const std::string& pluginName, std::string& parameter) const = 0;
+    /**
+     * @brief     Create a display path publisher
+     * @param[in] topic the topic
+     */
+    virtual void createDisplayPathPublisher(const std::string& topic, bool flag) = 0;
+
+    /**
+     * @brief     Create a received request publisher
+     * @param[in] topic the topic
+     */
+    virtual void createReceivedRequestPublisher(const std::string& topic, bool flag) = 0;
+
+    /**
+     * @brief     Create a contacts publisher
+     * @param[in] topic the topic
+     */
+    virtual void createContactsPublisher(const std::string& topic, bool flag) = 0;
```

Comment:
Add documentation for `flag` input parameter. Also consider renaming it into something more self-documenting if possible.